package/gluon-core/luasrc/lib/gluon/check-site.lua

```diff
@@ -370,6 +378,19 @@ function M.need_array_of(path, array, required)
 	return M.need_array(path, function(e) M.need_one_of(e, array) end, required)
 end
 
+function M.need_array_elements_exclusive(path, a, b, required)
+	local val = need_type(path, 'table', required, 'be an array')
+	if not val then
+		return nil
+	end
+
+	if contains(val, a) and contains(val, b) then
+		config_error(conf_src(path), 'expected %s to contain either "%s" or "%s", but not both.', path_to_string(path), a, b)
```

Comment:
This message is not quite accurate - it sounds like one of the values is required, but actually not having either is fine, too. Also, it should use the `format()` helper to achieve correct formatting for non-string array elements.

How about the following?
```suggestion
		config_error(conf_src(path), 'expected %s to contain only one of the elements %s and %s, but not both.', path_to_string(path), format(a), format(b))
```

(side node: `need_value()` also doesn't handle this correctly, it uses `tostring()` where `format()` would be preferable)